api/resolvers/sub.js

```diff
@@ -315,6 +315,56 @@ export default {
       notifyTerritoryTransfer({ models, sub, to: user })
 
       return updatedSub
+    },
+    unarchiveTerritory: async (parent, { name, hash, hmac, ...data }, { me, models, lnd }) => {
+      if (!me) {
+        throw new GraphQLError('you must be logged in', { extensions: { code: 'UNAUTHENTICATED' } })
+      }
+
+      const oldSub = await models.sub.findUnique({ where: { name } })
+      if (!oldSub) {
+        throw new GraphQLError('sub not found', { extensions: { code: 'BAD_INPUT' } })
+      }
+      if (oldSub.status !== 'STOPPED') {
+        throw new GraphQLError('sub is not archived', { extensions: { code: 'BAD_INPUT' } })
+      }
+      if (oldSub.billingType === 'ONCE') {
+        // sanity check. this should never happen but leaving this comment here
+        // to stop error propagation just in case and document that this should never happen.
+        // #defensivecode
+        throw new GraphQLError('sub should not be archived', { extensions: { code: 'BAD_INPUT' } })
+      }
+
+      const billingCost = TERRITORY_PERIOD_COST(data.billingType)
+      const billPaidUntil = nextBilling(new Date(), data.billingType)
+      const cost = BigInt(1000) * BigInt(billingCost)
+      const newSub = { ...data, billPaidUntil, billingCost, userId: me.id, status: 'ACTIVE' }
+
+      await serializeInvoicable([
+        models.user.update({
+          where: {
+            id: me.id
+          },
+          data: {
+            msats: {
+              decrement: cost
+            }
+          }
+        }),
+        models.subAct.create({
+          data: {
+            subName: name,
+            userId: me.id,
+            msats: cost,
+            type: 'BILLING'
+          }
+        }),
+        models.sub.update({ where: { name }, data: newSub }),
+        oldSub.userId !== me.id ? models.territoryTransfer.create({ data: { subName: name, oldUserId: oldSub.userId, newUserId: me.id } }) : null
+      ].filter(q => !!q),
```

Comment:
Nice! This is a clean way to conditionally include a query. We use a different pattern elsewhere to do this `filter` thing but this is simpler.

It could even be simplified to:

`oldSub.userId !== me.id && models.territoryTransfer.create({ data: { subName: name, oldUserId: oldSub.userId, newUserId: me.id } })`

----------------

I think it might be worth adding this filter logic to `serialize` and `serializeInvoicable` so we can get this kind of behavior everywhere.

Reply:
Done in baf338a4

> I think it might be worth adding this filter logic to serialize and serializeInvoicable so we can get this kind of behavior everywhere.

Yep, I agree